modules.d/95rootfs-block/module-setup.sh

```diff
@@ -44,6 +44,9 @@ cmdline_rootfs() {
         printf " root=%s" "$(shorten_persistent_dev "$(get_persistent_dev "$_dev")")"
     fi
     _fstype="$(find_mp_fstype /)"
+    if [[ ${_fstype} == "zfs" ]]; then
+        printf " root=ZFS=%s" "$(findmnt -n -o SOURCE /)"
```

Comment:
in upstream we treat the `root=zfs:<dataset>` format as canonical so it'd be nice if this were that

Reply:
also, if the root dataset has spaces in it (and this is valid and supported), you need to replace them with pluses, i.e.
```
         local _root_ds="$(findmnt -n -o SOURCE /)" 
         printf " root=zfs:%s" "${_root_ds// /+}"
```

Reply:
I will do that.